qiskit_nature/problems/second_quantization/molecular/fermionic_op_builder.py

```diff
@@ -0,0 +1,121 @@
+    """
+    Builds a fermionic operator based on 1- and/or 2-body integrals.
+    This method requires the integrals stored in the '*chemist*' notation
+             h2(i,j,k,l) --> adag_i adag_k a_l a_j
+    and the integral values are used for the coefficients of the second-quantized
+    Hamiltonian that is built. The integrals input here should be in block spin
+    format and also have indexes reordered as follows 'ijkl->ljik'
+    There is another popular notation, the '*physicist*' notation
+             h2(i,j,k,l) --> adag_i adag_j a_k a_l
+    If you are using the '*physicist*' notation, you need to convert it to
+    the '*chemist*' notation. E.g. h2=numpy.einsum('ikmj->ijkm', h2)
+    The :class:`~qiskit_nature.QMolecule` class has
+    :attr:`~qiskit_nature.QMolecule.one_body_integrals` and
+    :attr:`~qiskit_nature.QMolecule.two_body_integrals` properties that can be
+    directly supplied to the `h1` and `h2` parameters here respectively.
+
+    Args:
+        one_body_integrals (numpy.ndarray): One-body integrals stored in the chemist notation.
+        two_body_integrals (numpy.ndarray): Two-body integrals stored in the chemist notation.
+
+    Returns:
+        FermionicOp: FermionicOp built from 1- and/or 2-body integrals.
+    """
+
+    fermionic_op = _build_fermionic_op(one_body_integrals, two_body_integrals)
+    fermionic_op = fermionic_op.reduce()
+
+    return fermionic_op
+
+
+def _build_fermionic_op(one_body_integrals: np.ndarray,
+                        two_body_integrals: np.ndarray) -> FermionicOp:
+    one_body_base_ops = _create_one_body_base_ops(one_body_integrals)
+    two_body_base_ops = _create_two_body_base_ops(
+        two_body_integrals) if two_body_integrals is not None else []
+    base_ops = one_body_base_ops + two_body_base_ops
+
+    fermionic_op = FermionicOp('I' * len(one_body_integrals))
+    for base_op in base_ops:
+        fermionic_op += base_op
+    return fermionic_op
+
+
+def _create_one_body_base_ops(one_body_integrals: np.ndarray):
+    return _create_base_ops(one_body_integrals, 2, _calc_coeffs_with_ops_one_body)
+
+
+def _create_two_body_base_ops(two_body_integrals: np.ndarray):
+    return _create_base_ops(two_body_integrals, 4, _calc_coeffs_with_ops_two_body)
+
+
+def _create_base_ops(integrals: np.ndarray, repeat_num: int, calc_coeffs_with_ops):
+    base_ops_list = []
+    integrals_length = len(integrals)
+    for idx in itertools.product(range(integrals_length), repeat=repeat_num):
+        coeff = integrals[idx]
+        if not coeff:
+            continue
+        coeffs_with_ops = calc_coeffs_with_ops(idx)
+        base_op = _create_base_op_from_labels(coeff, integrals_length, coeffs_with_ops)
+        base_ops_list.append(base_op)
+    return base_ops_list
+
+
+def _calc_coeffs_with_ops_one_body(idx):
+    return [(idx[0], '+'), (idx[1], '-')]
+
+
+def _calc_coeffs_with_ops_two_body(idx):
+    return [(idx[0], '+'), (idx[2], '+'), (idx[3], '-'), (idx[1], '-')]
```

Comment:
I pass these functions as arguments so that I only have one `_create_base_ops_labels` function:
`_create_base_ops_labels(one_body_integrals, repeat_num, _calc_coeffs_with_ops_one_body)`
`_create_base_ops_labels(two_body_integrals, repeat_num, _calc_coeffs_with_ops_two_body)`
It helps me to avoid code duplication. Note how I pass them as functions, not as values they return.

Reply:
Okay but I still don't understand why you need to wrap these lists into functions. You could just pass the list as the argument in each case.

Reply:
`idx` does not exist at the moment of passing the list as an argument.
There is an alternative way, namely passing the whole iterator as an argument and getting rid of functions that you address but not sure if one way is clearly better than the other.

Reply:
Oh sorry I completely overlooked that `idx` is an input argument here. My bad